twig/contracts/registry.v.py

```diff
@@ -0,0 +1,126 @@
+# todo
+# get all versions of a package
+
+# Events
+Release: event({_package: indexed(bytes32), _version: bytes32, _uri: bytes32})
+
+owner: public(address)
+
+# Package Data: (package_id => value)
+packages: public(
+    {exists: bool, created_at: timestamp, updated_at: timestamp, name: bytes32}[bytes32]
+)
+
+#  Release Data: (release_id => value)
+releases: public(
+    {
+        exists: bool,
+        created_at: timestamp,
+        package_id: bytes32,
+        version: bytes32,
+        uri: bytes32,
+    }[bytes32]
+)
+
+
+package_count: public(int128)
+release_count: public(int128)
+package_ids: bytes32[128]
```

Comment:
Very curious what the vyper pattern would be to allow arbitrary numbers of releases.  Can we just use an arbitrarily large number here instead of  `128`?

Reply:
Wanted to work on a stack widget, which could solve this. Constant time insertions and accesses. Still not sure if it should be a full double-linked list ($$$ gas) or a very simple push/pop data structure with no ability to pop anything other than the end.

Reply:
@njgheorghita (maybe @jacqueswww or @fubuloubu can confirm)

What about this API which is in the vyper by example section:

https://vyper.readthedocs.io/en/latest/vyper-by-example.html#crowdfund

```python
package_ids: {id: bytes32}[int128]
```

Wouldn't that allow you to have an *effectively* infinite psuedo-array of them?

Reply:
Also, I assume it should be possible to do a mapping to a non-object type such as `int128 => bytes32` but it wasn't clear what that syntax would be...

Reply:
Yup, that would be the most efficient possibility. You would have to manage a separate "length" parameter to keep track of what the last entry you set was though.

A stack type widget could manage that for you under the hood. Same with a queue type.